File: backend/src/models/investors_model.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, HttpUrl


class Investor(BaseModel):
    id: int
    iinvestor_name: str
    is_parent: bool
    bqr_date: datetime
    country_of_risk_code: str
    investor_type_id: int
    aum_usd: float
    aum_source_link: HttpUrl
    aum_date: datetime
    nav_usd: float
    nav_source_link: HttpUrl
    nav_date: datetime
    ien_number: str
    is_national: int  # If this should be a boolean, change to bool
    investor_comment: Optional[str] = None
    created_by: str
    created_at: datetime

File: client-app/src/model/investors.model.ts
export interface Investor {
  id: number;
  iinvestor_name: string;
  is_parent: boolean;
  bqr_date: string;
  country_of_risk_code: string;
  investor_type_id: number;
  aum_usd: number;
  aum_source_link: string;
  aum_date: string;
  nav_usd: number;
  nav_source_link: string;
  nav_date: string;
  ien_number: string;
  is_national: number;
  investor_comment: string;
  created_by: string;
  created_at: string;
}
